framework/PostProcessors/DataMining.py

```diff
@@ -481,64 +485,41 @@ def collectOutput(self, finishedJob, outputObject):
-    ## End data copy
-    ############################################################################
-
-    ############################################################################
-    ## Now augment the DataObject by adding the computed labels and/or
-    ## transformed coordinates aka the new columns added to the DataObject
-    for key in dataMineDict['outputs']:
-      if key in missingKeys:
-        missingKeys.remove(key)
-      for param in outputObject.getVars('output'):
-        if key == param:
-          outputObject.remove(variable=key)
-      if outputObject.type == 'PointSet':
-        outputObject.addVariable(key, copy.copy(dataMineDict['outputs'][key]),classify='output')
-      elif outputObject.type == 'HistorySet':
-        expValues = np.zeros(len(outputObject), dtype=object)
-        values = dataMineDict['outputs'][key]
-        for index, value in enumerate(values):
-          timeLength = len(self.pivotVariable[index])
-          arrayBase = value * np.ones(timeLength)
-          xrArray = xr.DataArray(arrayBase,dims=(self.pivotParameter), coords=[self.pivotVariable[index]])
-          expValues[index] = xrArray
-        outputObject.addVariable(key, expValues,classify='output')
-    ## End data augmentation
     ############################################################################
+    if inputObject.type != outputObject.type:
+      self.raiseAnError(IOError,"The type of output DataObject",outputObject.name,"is not consistent with input",\
+              "DataObject type, i.e. ",outputObject.type,"!=",inputObject.type)
+    rlzs = {}
+    dataset = inputObject.asDataset().copy(deep=True)
+    sampleTag = inputObject.sampleTag
+    sampleCoord = dataset[sampleTag].values
+    availVars = dataset.data_vars.keys()
+    if outputObject.type == 'PointSet':
+      for key,value in dataMineDict['outputs'].items():
+        if key in availVars and not np.array_equal(value,dataset[key].values):
+          newDA = xr.DataArray(value,dims=(sampleTag),coords={sampleTag:sampleCoord})
+          dataset = dataset.drop(key)
+          dataset[key] = newDA
+        elif key not in availVars:
+          newDA = xr.DataArray(value,dims=(sampleTag),coords={sampleTag:sampleCoord})
+          dataset[key] = newDA
+    elif outputObject.type == 'HistorySet':
+      for key,values in dataMineDict['outputs'].items():
+        if key not in availVars:
+          expDict = {}
+          for index, value in enumerate(values):
+            timeLength = len(self.pivotVariable[index])
+            arrayBase = value * np.ones(timeLength)
+            xrArray = xr.DataArray(arrayBase,dims=(self.pivotParameter), coords=[self.pivotVariable[index]])
+            expDict[sampleCoord[index]] = xrArray
+          ds = xr.Dataset(data_vars=expDict)
+          ds = ds.to_array().rename({'variable':sampleTag})
+          dataset[key] = ds
```

Comment:
There's no comments to explain what's going on in this block, and I'm struggling to intuitively understand what's happening. Could you add some commentary?

Reply:
updated